### PR TITLE
[select] Fix QueryList when createNewItemPosition is first

### LINKED ISSUE #5773: queryList active item breaks when createNewItemPosition set to first
#### Environment

- __Package version(s)__: @blueprintjs/select@4.5.3
- __Operating System__: Windows 10
- __Browser name and version__: Chrome latest

#### Steps to reproduce

1. View select2 on docs website https://blueprintjs.com/docs/#select/select2
2. Enable "Allow creating new items"
3. Enable "Create new positions: first"
4. Open select and type "spirited away"

#### Actual behavior

The menu item exactly matching the query is  not active, and a duplicate is created when  pressing the enter key.

#### Expected behavior

The menu item exactly matching the query should be active, and become selected upon pressing the enter key. Same as if Create new position: first is disabled.

#### Possible solution

Checking `isCreateItemRendered()` checks the current value in state instead of the newly created `createNewItem`

https://github.com/palantir/blueprint/blob/2460933e7084632f9f8b4f193fcda9fb9b3fd38f/packages/select/src/components/query-list/queryList.tsx#L330-L336

https://github.com/palantir/blueprint/blob/2460933e7084632f9f8b4f193fcda9fb9b3fd38f/packages/select/src/components/query-list/queryList.tsx#L593-L602

Simple fix is to change `wouldCreatedItemMatchSomeExistingItem` to check the `createNewItem` created locally in `setQuery`, and use the value in state elsewhere.

### PR DESCRIPTION
**Pull request by jacobbel**

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/blueprint](https://github.com/palantir/blueprint), @jacobbel! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: packages/select/src/components/query-list/queryList.tsx [hunk 1]
```diff
@@ -561,14 +561,14 @@ export class QueryList<T> extends AbstractComponent2<QueryListProps<T>, IQueryLi
         return getFirstEnabledItem(this.state.filteredItems, this.props.itemDisabled, direction, startIndex);
     }
 
-    private isCreateItemRendered(): boolean {
+    private isCreateItemRendered(createNewItem?: T): boolean {
```
Comment (adidahiya) [inline packages/select/src/components/query-list/queryList.tsx]: can you please leave a `/** jsdoc */` comment on this method explaining briefly why we need this argument?

**Review by adidahiya — CHANGES_REQUESTED** [review]
nice, you seem to have fixed the issue, verified in the [docs preview build](https://output.circle-artifacts.com/output/job/c67a6dc0-e7ea-4bf8-8f9a-a5608fe22b19/artifacts/0/packages/docs-app/dist/index.html#select/select2):



can you please resolve the merge conflicts and add a quick code comment? thanks @jacobbel!

File: packages/select/src/components/query-list/queryList.tsx (hunk 1 above)
Comment (jacobbel) [reply packages/select/src/components/query-list/queryList.tsx]: All set 👍🏻

**Review by adidahiya — APPROVED** [review]

**Pull request merged**